traitsui/ui.py

```diff
@@ -264,7 +264,7 @@ def finish ( self ):
         """
 
         # Reset the contents of the user interface
-        self.reset( destroy = True )
+        self.reset( destroy = False )
```

Comment:
I think it should work. The wx `DestroyChildren` doc says it is automatically called by the destructor. @RobinD42 may have a better insight